commit 5e84dd547bce965c203cae71418cacb8592a0d3f
Author: Michael Ellerman <mpe@ellerman.id.au>
Date:   Tue Jan 21 15:29:59 2020 +1100

    powerpc/configs/skiroot: Enable some more hardening options
    
    Enable more hardening options.
    
    Note BUG_ON_DATA_CORRUPTION selects DEBUG_LIST and is essentially just
    a synonym for it.
    
    DEBUG_SG, DEBUG_NOTIFIERS, DEBUG_LIST, DEBUG_CREDENTIALS and
    SCHED_STACK_END_CHECK should all be low overhead and just add a few
    extra checks.
    
    SLAB_FREELIST_RANDOM, and SLUB_DEBUG_ON will add some overhead to the
    SLAB allocator, but nothing that should be meaningful for skiroot.
    
    Unselecting SLAB_MERGE_DEFAULT causes the SLAB to use more memory, but
    the skiroot kernel shouldn't be memory constrained on any of our
    systems, all it does is run a small bootloader.
    
    Disabling merging has some security/robustness benefit as it means a
    user-after-free or overflow will be limited to the objects in that
    slab, rather than potentially affecting objects from unrelated slabs
    that have been merged.
    
    Note also that slab merging is disabled anyway by enabling
    SLUB_DEBUG_ON, because of the SLAB_NEVER_MERGE mask.
    
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Acked-by: Joel Stanley <joel@jms.id.au>
    Link: https://lore.kernel.org/r/20200121043000.16212-9-mpe@ellerman.id.au

diff --git a/arch/powerpc/configs/skiroot_defconfig b/arch/powerpc/configs/skiroot_defconfig
index b05769d05fd6..1b6bdad36b13 100644
--- a/arch/powerpc/configs/skiroot_defconfig
+++ b/arch/powerpc/configs/skiroot_defconfig
@@ -23,6 +23,8 @@ CONFIG_EXPERT=y
 # CONFIG_AIO is not set
 CONFIG_PERF_EVENTS=y
 # CONFIG_COMPAT_BRK is not set
+# CONFIG_SLAB_MERGE_DEFAULT is not set
+CONFIG_SLAB_FREELIST_RANDOM=y
 CONFIG_SLAB_FREELIST_HARDENED=y
 CONFIG_PPC64=y
 CONFIG_ALTIVEC=y
@@ -294,6 +296,8 @@ CONFIG_LIBCRC32C=y
 # CONFIG_XZ_DEC_SPARC is not set
 CONFIG_PRINTK_TIME=y
 CONFIG_MAGIC_SYSRQ=y
+CONFIG_SLUB_DEBUG_ON=y
+CONFIG_SCHED_STACK_END_CHECK=y
 CONFIG_DEBUG_STACKOVERFLOW=y
 CONFIG_PANIC_ON_OOPS=y
 CONFIG_SOFTLOCKUP_DETECTOR=y
@@ -302,6 +306,10 @@ CONFIG_HARDLOCKUP_DETECTOR=y
 CONFIG_BOOTPARAM_HARDLOCKUP_PANIC=y
 CONFIG_WQ_WATCHDOG=y
 # CONFIG_SCHED_DEBUG is not set
+CONFIG_DEBUG_SG=y
+CONFIG_DEBUG_NOTIFIERS=y
+CONFIG_BUG_ON_DATA_CORRUPTION=y
+CONFIG_DEBUG_CREDENTIALS=y
 # CONFIG_FTRACE is not set
 CONFIG_XMON=y
 # CONFIG_RUNTIME_TESTING_MENU is not set